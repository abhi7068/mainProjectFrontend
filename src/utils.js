import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
