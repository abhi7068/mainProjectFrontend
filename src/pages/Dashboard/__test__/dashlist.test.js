import React from 'react';
import { render } from '../../../utils';
import Dashitems from '../index';

it('it renders', () => {
  render(<Dashitems />);
});
it('snapshot', () => {
  const { asFragment } = render(<Dashitems />);

  expect(asFragment(<Dashitems />)).toMatchSnapshot();
});
