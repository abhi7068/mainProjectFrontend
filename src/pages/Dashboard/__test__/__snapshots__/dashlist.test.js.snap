// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-151"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorPrimary MuiPaper-elevation4"
      style="background-color: black;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        style="background-color: black; display: flex; justify-content: flex-end;"
      >
         
        <ul
          class="MuiList-root MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root"
              style="color: white;"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Logout
              </span>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
             
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-155 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-padding"
        >
          <ul
            class="MuiList-root MuiList-padding"
          >
            <div
              class="MuiListItemIcon-root"
              style="color: white;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Admin
                </span>
              </div>
            </div>
          </ul>
        </ul>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding"
        >
          <div>
            <a
              href="/admin/dashboard"
              style="color: white;"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    style="color: white;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                    />
                  </svg>
                </div>
                Dashboard
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root"
                style="color: white;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
                style="color: white;"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Products
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              style="color: white;"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root"
                style="color: white;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 18.54l-7.37-5.73L3 14.07l9 7 9-7-1.63-1.27-7.38 5.74zM12 16l7.36-5.73L21 9l-9-7-9 7 1.63 1.27L12 16z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Categories
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                style="color: white;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <a
              href="/admin/list-users"
              style="color: white;"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root"
                  style="color: white;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                    />
                  </svg>
                </div>
                Users
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
